### PR TITLE
Add RenderStateRoot decorator to switch best practices stories

### PR DESCRIPTION
**Pull request by marianmwang**
## Summary:
- Switch best practices stories were not rendering bc a the `useUniqueId` hook needs the component to be a descendant of `RenderStateRoot`

Issue: N/A

## Test plan:
- `yarn start` and verify that the switch best practices stories are displaying

### CODE REVIEW

File: __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx [hunk 1]
```diff
@@ -31,6 +31,13 @@ export default {
             disableSnapshot: true,
         },
     },
+    decorators: [
+        (Story) => (
+            <RenderStateRoot>
+                <Story />
+            </RenderStateRoot>
+        ),
+    ],
```
Comment (jandrade) [inline __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx:40]: **question:** What do you think about adding this globally to all stories in `./storybook/preview.tsx`?

Your solution works fine for the Switch case, if you prefer to keep this PR only scoped for switch.

Comment (somewhatabstract) [reply __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx:40]: Make sure every story works in that case. By default, nested `RenderStateRoot` instances will throw an error so any story that is explicitly rendering one is going to break if you wrap them all by default. A story that is specifically using `RenderStateRoot` as an example but doesn't include it in the story would be weird, as would one that sets the `throwIfNested` value to `false` but we wanted to give a default, recommended usage story.

Comment (marianmwang) [reply __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx]: i took some inspiration from here https://github.com/storybookjs/storybook/issues/23237#issuecomment-1611351405 and added a global param that defaults to true. let me know you guys are ok w this solution or would prefer i just refactor the stories.

**Comment by somewhatabstract** [comment]
Hmm, none of these changes should affect code coverage stats. I think we need to add stories to the ignored files for code coverage gathering.

File: __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx (hunk 1 above)
Comment (somewhatabstract) [reply __docs__/wonder-blocks-switch/switch-best-practices.stories.tsx]: Yeah, that's a good compromise, @marianmwang - I like that. :)

**Comment by jandrade** [comment]
@marianmwang side note re: Snapshot releases:

There's something odd with the script... As far as I can see this PR shouldn't include any release changes, but I checked that the GH workflow step says otherwise:

```
🦋  @khanacademy/wonder-blocks-banner@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-birthday-picker@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-dropdown@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-icon-button@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-modal@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-popover@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-search-field@0.0.0-PR2050-20230911180742
🦋  @khanacademy/wonder-blocks-tooltip@0.0.0-PR2050-20230911180742
```

**Review by jandrade — APPROVED** [review]
Makes sense to me! thanks for trying out the global approach and finding a suitable solution 🚢

**Comment by marianmwang** [comment]
> @marianmwang side note re: Snapshot releases:
> 
> There's something odd with the script... As far as I can see this PR shouldn't include any release changes, but I checked that the GH workflow step says otherwise:
> 
> ```
> 🦋  success packages published successfully:
> 🦋  @khanacademy/wonder-blocks-banner@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-birthday-picker@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-dropdown@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-icon-button@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-modal@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-popover@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-search-field@0.0.0-PR2050-20230911180742
> 🦋  @khanacademy/wonder-blocks-tooltip@0.0.0-PR2050-20230911180742
> ```
> 
> Do you know why this could be happening?
> 
> For example, I checked in npm and the banner snapshot release got published: https://www.npmjs.com/package/@khanacademy/wonder-blocks-banner/v/0.0.0-PR2050-20230911180742)

yes @jandrade, so i believe the issue is that the script checks if there are any changesets **at all** in this branch by using `yarn changeset status`, which means it will catch any that are in the base branch and not versioned yet. i think it would be preferable to only add tags for packages that are changed in this PR (like the changeset bot does). do you want me to make a ticket for that?

**Comment by jandrade** [comment]
> > @marianmwang side note re: Snapshot releases:
> > There's something odd with the script... As far as I can see this PR shouldn't include any release changes, but I checked that the GH workflow step says otherwise:
> > ```
> > 🦋  success packages published successfully:
> > 🦋  @khanacademy/wonder-blocks-banner@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-birthday-picker@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-dropdown@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-icon-button@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-modal@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-popover@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-search-field@0.0.0-PR2050-20230911180742
> > 🦋  @khanacademy/wonder-blocks-tooltip@0.0.0-PR2050-20230911180742
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do you know why this could be happening?
> > For example, I checked in npm and the banner snapshot release got published: https://www.npmjs.com/package/@khanacademy/wonder-blocks-banner/v/0.0.0-PR2050-20230911180742)
> 
> yes @jandrade, so i believe the issue is that the script checks if there are any changesets **at all** in this branch by using `yarn changeset status`, which means it will catch any that are in the base branch and not versioned yet. i think it would be preferable to only add tags for packages that are changed in this PR (like the changeset bot does). do you want me to make a ticket for that?

ahh ok, that's a good question. @jeremywiebe tagging you to see if you dealt with similar issues in Perseus and which approach did you decided to take.

**Comment by jeremywiebe** [comment]
> > > @marianmwang side note re: Snapshot releases:
> > > There's something odd with the script... As far as I can see this PR shouldn't include any release changes, but I checked that the GH workflow step says otherwise:
> > > ```
> > > 🦋  success packages published successfully:
> > > 🦋  @khanacademy/wonder-blocks-banner@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-birthday-picker@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-dropdown@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-icon-button@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-modal@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-popover@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-search-field@0.0.0-PR2050-20230911180742
> > > 🦋  @khanacademy/wonder-blocks-tooltip@0.0.0-PR2050-20230911180742
> > > ```
> > > 
> > >   
> > > Do you know why this could be happening?
> > > For example, I checked in npm and the banner snapshot release got published: https://www.npmjs.com/package/@khanacademy/wonder-blocks-banner/v/0.0.0-PR2050-20230911180742)
> > 
> > 
> > yes @jandrade, so i believe the issue is that the script checks if there are any changesets **at all** in this branch by using `yarn changeset status`, which means it will catch any that are in the base branch and not versioned yet. i think it would be preferable to only add tags for packages that are changed in this PR (like the changeset bot does). do you want me to make a ticket for that?
> 
> ahh ok, that's a good question. @jeremywiebe tagging you to see if you dealt with similar issues in Perseus and which approach did you decided to take.

I don't know that I agree. Let's say we have the following PRs (arrows representing pointers to base PRs): `main <- A <- B`. 

Now we make some changes in PR `A` to package `wonder-blocks-banner`.  Then in PR `B` we make changes to the `wonder-blocks-search` (which uses changes to `wonder-blocks-banner` from PR `A`).  If we don't version/snapshot the `wonder-blocks-banner` package with the changes from PR `A`, we'll have an inconsistent set of packages. This is true even for PRs that have `main` as the base branch but depend on unreleased changes. 

We _could_ do possibly do more detailed dependency analysis, but for any given PR, right now it seems safest to version all packages that have unreleased changes.

**Pull request merged**